### DISCUSSION TITLE
how i can use another type  @this.set('description', description)

### DISCUSSION THREAD

**Discussion by vahidalvandi** [livewire/livewire#5089] [1. Help] [answered]
i need use another type in no blade  `@this.set('description', description) `to  none blade
after compile change to like this

      window.livewire.find('cCakXW7NU6Ohm63GmX4d').set('description', description)

Note: the @this directive compiles to the following string for JavaScript to interpret: "window.livewire.find([component-id])"

**Comment by Martleby** [comment]
I may misunderstand you, but I think you're asking how to get a component's id in an inline script, but you can't (or aren't) using Blade and so cannot use @this?

Off the top of my head, you could put an id="yourid" on the component's div and then:

```
let wireId = document.getElementById('yourid').getAttribute('wire:id')
window.livewire.find(wireId).set('description', description)
``` 

There may be a better way, but that seemed the most obvious to me.

Cheers,
Martin.

**Reply by vahidalvandi** [reply]
Do have better way ? 
After encode blade with source guardian or ioncube need it cannot convert to none blade .

**Reply by joshhanley** [reply]
@vahidalvandi I'm not sure I follow, what do you mean by once you encode it, it won't work? What does `@this` look like after it has been encoded?

**Reply by vahidalvandi** [reply]
> @vahidalvandi I'm not sure I follow, what do you mean by once you encode it, it won't work? What does `@this` look like after it has been encoded?

after encoded @this not work and not detect to a component

**Reply by joshhanley** [reply]
@vahidalvandi can you share an example of it looks like once it has been encoded?

Also can you share your code where you want to replace `@this` so we have some context so we might be able to assist you?

Thanks!

**Comment by Martleby** [comment, marked as answer]
Ah, I think I understand. You're using a 3rd party tool to encrypt/protect your php which stops Liverwire's blade helpers from working?

I had a look at the source code to find out what @this is really doing (https://github.com/livewire/livewire/blob/master/src/LivewireBladeDirectives.php):

```
return "window.livewire.find('{{ \$_instance->id }}')";
```

So I think you should be able to do window.livewire.find('{{ \$_instance->id }}') instead of @this.

A word of caution: this might work, but if the implementation of @this ever changes you might be in for a world of pain.

**Reply by vahidalvandi** [reply]
@Martleby thank you

like bellow code ?
`  window.livewire.find('{{ $_instance->id }}').set('description', description)
`

**Reply by vahidalvandi** [reply]
work fine thank you
